torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2272,6 +2288,49 @@ def _make_tensor_helper(shape, low=None, high=None):
                        SkipInfo('TestCommon', 'test_variant_consistency_jit',
                                 device_type='cuda', dtypes=[torch.float16]),
                    )),
+    UnaryUfuncInfo('sign',
+                   ref=np.sign,
+                   dtypes=all_types_and(torch.bfloat16, torch.half),
+                   dtypesIfCPU=all_types_and(torch.bool, torch.bfloat16, torch.half),
+                   dtypesIfCUDA=all_types_and(torch.bool, torch.bfloat16, torch.half),
+                   skips=(
+                       # Reference: https://github.com/pytorch/pytorch/issues/41245
```

Comment:
Should we take this opportunity to fix this discrepancy?

Reply:
I would prefer to do it in a seperate PR as we will have to update the vectorized path with checks for Nan (which would regress the performance slightly) and add relevant benchmark.

https://github.com/pytorch/pytorch/blob/3ce51fd5f49140b7c62c7cb3585d30566ad07dcf/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L268-L289

Does that sound ok?

Reply:
Separate PR sounds good.